public/app/core/components/PageHeader/PageHeader.tsx

```diff
@@ -76,19 +76,28 @@ const Navigation = ({ children }: { children: NavModelItem[] }) => {
 export const PageHeader: FC<Props> = ({ model }) => {
   const styles = useStyles2(getStyles);
 
+  const main = model.main;
+  const [children, setChildren] = useState(main.children);
+
+  useEffect(() => {
+    const updatedChildren = main.children?.map((child) => {
+      // Add suffix component for the items that come from backend and are highlighted
+      if (child.highlightText && !child.tabSuffix) {
+        return {
+          ...child,
+          tabSuffix: () =>
+            ProBadge({ experimentId: child.highlightId ? `feature-highlights-${child.highlightId}-badge` : '' }),
+        };
+      }
+      return child;
+    });
+    setChildren(updatedChildren);
+  }, [main.children]);
```

Comment:
Do we really need this badge event? We already have an onclick event when users click the recorded queries tab, seems like the badge event is only to detect if the users have seen the tab (which is quite a questionable metric in this case since it'll be emitted anytime they go to config page), but if it's really necessary we could maybe add this event to the whole config page and trigger it when feature highlights are enabled? We probably should consider the same approach for ProBadge in other places, since that would make badge-related code simpler.